coffee_httpd/src/httpd/server.rs

```diff
@@ -64,6 +69,22 @@ async fn coffee_help(
     Ok(body)
 }
 
+#[api_v2_operation]
+#[get("/list")]
+async fn coffee_list(data: web::Data<AppState>) -> Result<Json<Value>, Error> {
+    let mut coffee = data.coffee.lock().await;
+    let result = coffee.list().await;
+    match result {
+        Ok(coffee_list) => {
+            let plugins: Vec<Plugin> = coffee_list.plugins;
+            Ok(Json(json!({ "Plugins": plugins })))
```

Comment:
```suggestion
            Ok(Json(json!({ "plugins": coffee_list.plugins })))
```

Reply:
Ah no, this is wrong, and your code too.

There is no need to use the `json` macro; you can use the value directly
`Json(coffee_list)` or if the Json is not so smart, you can do `Json(serde_json::to_value(coffee_list))`

The `coffee_list` is already the correct format that we want

Reply:
In this case our return value would be of type `Json<CoffeeList>` which causes errors
We would have to implement the `TypedData` and `Apiv2Schema` traits for `CoffeeList` type.

Reply:
We can do this
```
let result = coffee.list().await;
match result {
      Ok(coffee_list) => match serde_json::to_value(coffee_list) {
          Ok(val) => Ok(Json(val)),
          Err(_) => Err(actix_web::error::ErrorInternalServerError(
              "coffee list error",
          )),
      },
      Err(_) => Err(actix_web::error::ErrorInternalServerError(
          "coffee list error",
      )),
```
and the return value remain of type `Json<Value>`
Is this better than the current changes?

Reply:
yes it is better

Reply:
but please do not give the coffee error

```
let val = serde_json::to_value(coffee_list)
            .map_err(|err| actix_web::error::ErrorInternalServerError(format!("coffee list error: {err}")))?;

```

if this do not work, implement the `std::fmt::Display` on the `CoffeeError`

Reply:
Works fine. I tested it as well